PinClustering/PinClustering/ViewControllers/MapViewController.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  MapViewController.swift
+//  PinClustering
+//
+//  Created by Norbert Agoston on 06/09/16.
+//  Copyright © 2016 Norbert Agoston. All rights reserved.
+//
+
+import UIKit
+import MapKit
+
+class MapViewController: UIViewController {
+
+    @IBOutlet weak var map: MKMapView!
+    
+    var pointsOfInterest = [AnyObject]()
+    
+    var town: Town? {
+        didSet {
+            loadLocations()
+        }
+    }
+    
+    // MARK: Lifecycle
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+    }
+    
+    override func viewWillAppear(animated: Bool) {
+        super.viewWillAppear(animated)
+        navigationController?.navigationBarHidden = false
+    }
+    
+    // MARK: Private
+    
+    func loadLocations() {
+        guard let url = town?.webServiceURL else {
+            return
+        }
+
+        NetworkManager.sharedInstance.httpGet(url) { (locations, error) in
+            guard let actualLocations = locations else {
+                if let error = error {
+                    AlertManager.sharedInstance.displayAlert("Error retrieving locations", message: error, presentingViewController: self)
+                }
+                return
+            }
+            
+            self.updatePointsOfInterestWith(actualLocations)
+            dispatch_async(dispatch_get_main_queue()) { [unowned self] in
```

Comment:
return every callback from the network manager on main thread so you don't have to do this in every controller. It's prone to error if you do it like this. For example the AlertManager.sharedInstance.displayAlert won't execute on the main thread and it will delay the appearance of the alert.